src/_pytest/python_api.py

```diff
@@ -406,7 +406,11 @@ def __repr__(self) -> str:
         # If a sensible tolerance can't be calculated, self.tolerance will
         # raise a ValueError.  In this case, display '???'.
         try:
-            vetted_tolerance = f"{self.tolerance:.1e}"
+            if self.tolerance >= 1e-3 and self.tolerance < 1e3:
```

Comment:
Same thing, but a bit more Pythonic:

```suggestion
            if 1e-3 <= self.tolerance < 1e3:
```